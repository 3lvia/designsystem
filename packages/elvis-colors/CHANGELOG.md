# Elvia Colors Changelog

## 1.7.0 (27.02.23)

### New feature

- Added new theme color labels for icon colors.

### Patch

- Added missing export for `getThemeColorContrast()`.

## 1.6.0 (24.02.23)

### New feature

- Added new theme color labels: `background-overlay-strong`, `state-selected-grey`,
  `state-disabled-foreground`, `state-disabled-background`, `state-disabled-foreground-strong`,
  `state-disabled-background-strong`, `static-black`, and `static-white`.
- Updated color of `background-accent-strong` in dark theme.
- Now generating css variables for contrast colors for each theme color.
- Added new function `getThemeColorContrast()` to get the contrast color for a theme color label.

## 1.5.0 (30.01.23)

### New feature

- Added theme functionality. This introduces new functions `getThemeColor()` and `getCustomThemeColor()` along
  with several new Typescript types related to theming.
- Deprecated the function `getColor()`.

## 1.4.3 (01.12.22)

### Patch

- The Typescript union type for all the color names is no longer generated during build, instead it comes
  directly from Typescript.
- The package has been rewritten in Typescript.
- The build script is now written in Typescript.

## 1.4.2 (29.08.22)

### Patch

- Now generating up a Typescript union type for all the color names.

## 1.4.1 (11.05.22)

### Patch

- Added generation of scss file for elvis (colorMap.scss).

## 1.4.0 (10.05.22)

### New feature

- Expanded the build system and added a dist folder.

## 1.3.2 (03.05.22)

### Bug fix

- Fixed type declaration of colors object.

## 1.3.1 (08.04.22)

### Patch

- Converted from ES module (ESM) to CommonJS (CJS).

## 1.3.0 (08.04.22)

### New feature

- Improved support for IntelliSense when using `getColor()` and `getContrastText()`.

## 1.2.3 (23.03.22)

### Patch

- Now building the elviaColors.json and elviaColors.scss files from the elviaColors.js colors object using
  gulp.

### Bug fixes

- Fixed the RGB value on signal color yellow.

## 1.2.2 (08.03.22)

### Patch

- Updated color alt-labels to correspond with the Colors documentation on design.elvia.io.

## 1.2.1 (19.01.22)

### Patch

- Updated README to explain how to use the package in Typescript/Javascript.

## 1.2.0 (07.01.22)

### New feature

- Added JSDoc documentation and better Typescript declarations.

## 1.1.0 (29.11.21)

### New feature

- Added new helper functions to get color hex value and contrast text from color label or alt-text.
