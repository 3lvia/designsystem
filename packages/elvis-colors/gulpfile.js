'use strict';

const fs = require('fs');
const gulp = require('gulp');
const prettier = require('gulp-prettier');
const header = require('gulp-header');
const del = require('del');
const colors = require('./src/elviaColors')['default'];

const WARNING = `// THIS FILE IS AUTOMATICALLY GENERATED AND WILL BE OVERWRITTEN.
// DO NOT MAKE CHANGES TO THIS FILE DIRECTLY.\n\n`;

/**
 * Makes a capitalized camel case name from a kebab case name.
 * @param {string} name Original name in kebab case.
 * @returns Capitalized camel case name.
 */
const makeCapitalizedCamelCase = (name) => {
  const arr = name.split('-');
  for (let i = 0; i < arr.length; i++) {
    arr[i] = arr[i].charAt(0).toUpperCase() + arr[i].slice(1);
  }
  return arr.join('');
};

/**
 * Rename a color name (or alt-label) from elviaColors.js to its scss name.
 *
 * If a name is prefixed with "elvis" or "elvia", this prefix is removed.
 *
 * Finally, all names are prefixed with "Elvia".
 * @param {string} name Original name.
 * @returns New name.
 */
const renameToScssName = (name) => {
  let returnName = 'Elvia';
  if (name.startsWith('elvis') || name.startsWith('elvia')) {
    returnName = returnName + makeCapitalizedCamelCase(name.substring(5));
  } else {
    returnName = returnName + makeCapitalizedCamelCase(name);
  }
  return returnName;
};

const generateElviaColorsJsonFile = async () => {
  const colorsJSON = JSON.stringify(colors);
  fs.writeFileSync('./dist/elviaColors.json', colorsJSON);
};

/**
 * Iterates through the colors object in elviaColors.js and adds each
 * color (and alt-label) to an array as a string in scss-format.
 * Will not produce any duplicate entries.
 *
 * Then writes the elviaColors.scss file containing all the entries from the array.
 */
const generateElviaColorsScssFile = async () => {
  let content = [];
  for (const categoryLabel in colors) {
    for (const colorLabel in colors[categoryLabel]) {
      // Each color is added with its color hex value.
      content.push(`$${renameToScssName(colorLabel)}: ${colors[categoryLabel][colorLabel]['color']};`);
      if (colors[categoryLabel][colorLabel]['alt-labels']) {
        for (const altLabelIndex in colors[categoryLabel][colorLabel]['alt-labels']) {
          const altLabel = colors[categoryLabel][colorLabel]['alt-labels'][altLabelIndex];
          // Alt-labels are added as references to their parent color.
          const newContent = `$${renameToScssName(altLabel)}: $${renameToScssName(colorLabel)};`;
          // Make sure the color isn't already in the array.
          if (content.indexOf(newContent) === -1) {
            content.push(newContent);
          }
        }
      }
    }
  }
  content.splice(2, 0, '$FontColorLight: $ElviaWhite;');
  fs.writeFileSync('./dist/elviaColors.scss', WARNING + content.join('\n') + '\n');
};

const copyElviaColors = async () => {
  return gulp
    .src(['./src/elviaColors.js', './src/elviaColors.d.ts'])
    .pipe(header(WARNING))
    .pipe(gulp.dest('./dist/'));
};

const formatPrettier = async () => {
  return gulp.src(['./dist**/*']).pipe(prettier()).pipe(gulp.dest('.'));
};

function cleanup() {
  return del(['./dist/**/*'], { force: true });
}

gulp.task(
  'default',
  gulp.series(
    cleanup,
    generateElviaColorsJsonFile,
    generateElviaColorsScssFile,
    copyElviaColors,
    formatPrettier,
    function (done) {
      console.log('Elvis-colors - Successfully built Elvis-colors! ');
      done();
    },
  ),
);
