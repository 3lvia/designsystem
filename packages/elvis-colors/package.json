{
  "name": "@elvia/elvis-colors",
  "version": "4.1.0",
  "description": "Elvia colors",
  "license": "MIT",
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "gulp && tsup && yarn sourcemap",
    "sourcemap": "tsc",
    "start": "tsup --watch"
  },
  "files": [
    "CHANGELOG.json",
    "README.md",
    "dist/**/*",
    "!dist/**/*.d.ts.map"
  ],
  "devDependencies": {
    "@types/gulp": "^4.0.17",
    "@types/node": "^20.11.4",
    "del": "^6.1.1",
    "gulp": "^4.0.2",
    "ts-node": "^10.9.2",
    "tsup": "^8.0.1",
    "typescript": "~5.2.2"
  }
}
