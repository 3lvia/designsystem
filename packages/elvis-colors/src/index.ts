export * from './getThemeColor';
export * from './theme';
export * from './themes/lightTheme';
export * from './themes/darkTheme';
export * from './shadows';
