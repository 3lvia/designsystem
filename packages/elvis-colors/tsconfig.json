{
  "extends": "../components/tsconfig.json",
  "ts-node": {
    "compilerOptions": {
      "module": "commonjs"
    }
  },
  "compilerOptions": {
    "declaration": true,
    "emitDeclarationOnly": true,
    "declarationMap": true,
    "outDir": "./dist/"
  },
  "files": ["src/index.ts"]
}
