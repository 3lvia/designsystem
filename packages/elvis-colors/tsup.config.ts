import { defineConfig } from 'tsup';

export default defineConfig((options) => ({
  name: '@elvia/elvis-colors',
  entry: ['src/index.ts'],
  format: ['cjs', 'esm'],
  sourcemap: true,
  minify: !options.watch,
  dts: true,
}));
