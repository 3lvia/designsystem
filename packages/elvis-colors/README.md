# Elvia colors

**This package is meant for internal use only. All colors are available through
[Elvis](https://www.npmjs.com/package/@elvia/elvis). This package should not be imported by an end-user
directly.**

Available colors reflect the colors illustrated at https://design.elvia.io/brand/color#Overview.

All Color variables are named Elvia< Name > in CamelCase. Eg ElviaBlack, ElviaOrangeMango.

## Usage SCSS

### Recommended @Use

Insert at top of your scss file :<br>

- ```scss
  @use '@elvia/elvis-colors';
  ```

If this does not work for you, you may need to specify:

- ```scss
  @use '@elvia/elvis-colors/dist/elviaColors.scss';
  ```

Example: <br>

- ```scss
  border: 1px solid elviaColors.$ElviaBlack;
  ```

### Alternative @Import

For making all variables accessible throughout your project you can use @Import. This function will be
deprecated by sass in the future and we therefore recommend applying @use instead.

Add colors file from package to scss file like: <br>

- ```scss
  @import '~@elvia/elvis-colors';
  ```

If this does not work for you, you may need to specify:

- ```scss
  @import '~@elvia/elvis-colors/dist/colors.scss';
  ```

Example: <br>

- ```scss
  border: 1px solid $ElviaBlack;
  ```

## Usage Typescript/Javascript

To use this package you can import the function `getThemeColor()`. It takes the label of the requested color
as input, and returns a css variable with a fallback to its its hex value if it exists in the Elvia colors.
<br>

- ```typescript
  import { getThemeColor } from '@elvia/elvis-colors';
  const color = getThemeColor('state-on');
  ```

Likewise, the function `getThemeColorContrast()` is provided to get a contrast text color corresponding to the
requested color.

A previous version of this package provided the function `getColor()`. This function is now deprecated and
will be removed in a future version. It takes the label of the requested color as input, and returns its hex
value.

- ```typescript
  import { getColor } from '@elvia/elvis-colors';
  const color = getColor('elvia-charge');
  ```

## Maintaining

The json and scss-files are generated from the colors object in the ts file. When making changes, remember to
run `yarn build` afterwards.
