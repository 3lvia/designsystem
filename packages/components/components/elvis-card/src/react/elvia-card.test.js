import Card from './elvia-card.tsx';
import React from 'react';
import { mount } from 'enzyme';
import { getColor } from '@elvia/elvis-colors';

describe('Elvis Card', () => {
  let wrapper;
  let cardArea;
  let cardColoredLine;
  let cardIcon;
  let cardHeader;
  let cardDescription;
  let cardLabel;
  let cardHoverArrow;
  let cardCornerIcon;

  describe('Type = simple square', () => {
    beforeEach(() => {
      wrapper = mount(
        <Card icon={'Icon'} header={'Header'} description={'Description'} label={'Label'}></Card>,
      );
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
      cardColoredLine = wrapper.find({ 'data-testid': 'card-colored-line' });
      cardIcon = wrapper.find({ 'data-testid': 'card-icon' });
      cardHeader = wrapper.find({ 'data-testid': 'card-header' });
      cardDescription = wrapper.find({ 'data-testid': 'card-description' });
      cardLabel = wrapper.find({ 'data-testid': 'card-label' });
      cardHoverArrow = wrapper.find({ 'data-testid': 'card-detail-hover-arrow' });
      cardCornerIcon = wrapper.find({ 'data-testid': 'card-corner-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should have icon', function (done) {
      expect(cardIcon.at(0).text()).toBe('Icon');
      done();
    });

    it('should have header', function (done) {
      expect(cardHeader.at(0).text()).toBe('Header');
      done();
    });

    it('should have description', function (done) {
      expect(cardDescription.at(0).text()).toBe('Description');
      done();
    });

    it('should have shape square', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border-radius: 8px`);
      done();
    });

    it('should have border', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border: 1px solid ${getColor('grey-10')}`);
      done();
    });

    it('should not have top border with color', function (done) {
      expect(cardColoredLine.length).toEqual(0);
      done();
    });

    // TODO: Figure out how to test css states (e.g. :hover)
    // it('should not have top border with color on hover', function (done) {
    //   wrapper.simulate('mouseenter');
    //   expect(cardColoredLine.at(0).getDOMNode()).not.toHaveStyle(`border-top: 4px solid ${getColor('red')}`);
    //   done();
    // });

    // it('should switch border on hover', function (done) {
    //   expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border: 1px solid ${getColor('grey-10')}`);
    //   //   wrapper.simulate('mouseenter');
    //   expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border: 2px solid ${getColor('elvia-charge')}`);
    //   done();
    // });

    it('should not have corner icon', function (done) {
      expect(cardCornerIcon.length).toEqual(0);
      done();
    });

    it('should not have hover arrow', function (done) {
      expect(cardHoverArrow.length).toEqual(0);
      done();
    });

    it('should not have label', function (done) {
      expect(cardLabel.length).toEqual(0);
      done();
    });

    it('should not switch icon on hover', function (done) {
      expect(cardIcon.at(0).text()).toBe('Icon');
      wrapper.simulate('mouseenter');
      expect(cardIcon.at(0).text()).toBe('Icon');
      done();
    });

    it('should have default width', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`width: 100%`);
      done();
    });

    it('should have default minWidth', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`min-width: 112px`);
      done();
    });

    it('should have default maxWidth', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`max-width: 400px`);
      done();
    });
  });

  describe('Type = simple square with width and colored top', () => {
    beforeEach(() => {
      wrapper = mount(
        <Card
          icon={'Icon'}
          header={'Header'}
          description={'Description'}
          borderColor={'red'}
          label={'Label'}
          width={'150px'}
          minWidth={150}
          maxWidth={350}
        ></Card>,
      );
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
      cardColoredLine = wrapper.find({ 'data-testid': 'card-colored-line' });
      cardIcon = wrapper.find({ 'data-testid': 'card-icon' });
      cardHeader = wrapper.find({ 'data-testid': 'card-header' });
      cardDescription = wrapper.find({ 'data-testid': 'card-description' });
      cardLabel = wrapper.find({ 'data-testid': 'card-label' });
      cardHoverArrow = wrapper.find({ 'data-testid': 'card-detail-hover-arrow' });
      cardCornerIcon = wrapper.find({ 'data-testid': 'card-corner-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should have top border with color', function (done) {
      expect(cardColoredLine.at(0).getDOMNode()).toHaveStyle(`border-top: 4px solid ${getColor('red')}`);
      done();
    });

    it('should have a set width', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`width: 150px`);
      done();
    });

    it('should have minWidth', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`min-width: 150px`);
      done();
    });

    it('should have maxWidth', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`max-width: 350px`);
      done();
    });
  });

  describe('Type = simple square, with hover icon', () => {
    beforeEach(() => {
      wrapper = mount(<Card icon={'Icon'} iconHover={'IconHover'} minWidth={50}></Card>);
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
      cardIcon = wrapper.find({ 'data-testid': 'card-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should switch icon on hover', function (done) {
      expect(cardIcon.at(0).text()).toBe('Icon');
      wrapper.simulate('mouseenter');
      expect(cardIcon.at(0).text()).toBe('IconHover');
      done();
    });
  });

  describe('Type = simple square, no border', () => {
    beforeEach(() => {
      wrapper = mount(<Card icon={'Icon'} hasBorder={false}></Card>);
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should not have standard border', function (done) {
      expect(cardArea.at(0).getDOMNode()).not.toHaveStyle(`border: 1px solid ${getColor('grey-10')}`);
      done();
    });

    it('should have grey border (to match "on grey")', function (done) {
      expect(cardArea.at(0).getDOMNode()).not.toHaveStyle(`border: 1px solid ${getColor('grey-10')}`);
      done();
    });
  });

  describe('Type = simple circle', () => {
    beforeEach(() => {
      wrapper = mount(
        <Card
          shape={'circle'}
          icon={'Icon'}
          header={'Header'}
          description={'Description'}
          borderColor={'red'}
          cornerIcon={'CornerIcon'}
        ></Card>,
      );
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
      cardColoredLine = wrapper.find({ 'data-testid': 'card-colored-line' });
      cardHoverArrow = wrapper.find({ 'data-testid': 'card-detail-hover-arrow' });
      cardCornerIcon = wrapper.find({ 'data-testid': 'card-corner-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should have shape circle', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border-radius: 50%`);
      done();
    });

    it('should not have top border with color', function (done) {
      expect(cardColoredLine.length).toEqual(0);
      done();
    });

    it('should not have corner icon', function (done) {
      expect(cardCornerIcon.length).toEqual(0);
      done();
    });

    it('should not have hover arrow', function (done) {
      expect(cardHoverArrow.length).toEqual(0);
      done();
    });
  });

  describe('Type = detail', () => {
    beforeEach(() => {
      wrapper = mount(
        <Card
          type={'detail'}
          label={'Label'}
          description={'Description'}
          cornerIcon={'CornerIcon'}
          shape={'circle'}
        ></Card>,
      );
      cardArea = wrapper.find({ 'data-testid': 'card-area' });
      cardIcon = wrapper.find({ 'data-testid': 'card-icon' });
      cardDescription = wrapper.find({ 'data-testid': 'card-description' });
      cardLabel = wrapper.find({ 'data-testid': 'card-label' });
      cardHoverArrow = wrapper.find({ 'data-testid': 'card-detail-hover-arrow' });
      cardCornerIcon = wrapper.find({ 'data-testid': 'card-corner-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should not have icon', function (done) {
      expect(cardIcon.length).toEqual(0);
      done();
    });

    it('should have label', function (done) {
      expect(cardLabel.at(0).text()).toBe('Label');
      done();
    });

    it('should have label with styling', function (done) {
      expect(cardLabel.at(0).getDOMNode()).toHaveStyle(
        `padding: 4px 8px;
        border-radius: 4px;
        background: ${getColor('grey-10')}; 
        font-size: 11px;
        font-weight: 400;`,
      );
      done();
    });

    it('should have shape square', function (done) {
      expect(cardArea.at(0).getDOMNode()).toHaveStyle(`border-radius: 8px`);
      done();
    });

    it('should have corner icon', function (done) {
      expect(cardCornerIcon.at(0).text()).toBe('CornerIcon');
      done();
    });

    it('should have hover arrow', function (done) {
      expect(cardHoverArrow.length).toEqual(2);
      done();
    });

    it('should have default lines of description text', function (done) {
      expect(cardDescription.at(0).getDOMNode()).toHaveStyle(`line-clamp: 5;`);
      done();
    });
  });

  describe('Type = detail, shorter description', () => {
    beforeEach(() => {
      wrapper = mount(
        <Card type={'detail'} label={'Label'} description={'Description'} maxDescriptionLines={3}></Card>,
      );
      cardDescription = wrapper.find({ 'data-testid': 'card-description' });
      cardCornerIcon = wrapper.find({ 'data-testid': 'card-corner-icon' });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('should have max 3 lines of description text', function (done) {
      expect(cardDescription.at(0).getDOMNode()).toHaveStyle(`line-clamp: 3;`);
      done();
    });

    it('should have empty corner icon', function (done) {
      expect(cardCornerIcon.at(0).text()).toEqual('');
      done();
    });
  });
});
