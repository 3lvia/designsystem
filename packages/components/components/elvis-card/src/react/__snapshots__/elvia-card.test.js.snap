// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elvis Card Snapshots Type = detail should render correctly 1`] = `
.c1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #fff;
  overflow: hidden;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 250px;
  max-width: 400px;
  padding: 24px;
  width: 100%;
  height: 100%;
  border-radius: 8px;
  border: 1px solid #e9e9e9;
}

.c1:hover {
  border: 2px solid #29d305;
  padding: 23px;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding-left: 0;
  padding-right: 0;
}

.c3 {
  font-family: "Red Hat Text",Verdana,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: left;
  color: #000;
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 5;
  line-clamp: 5;
  -webkit-box-orient: vertical;
}

.c5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border-radius: 4px;
  background: #e9e9e9;
  font-family: 'Red Hat Text','Verdana, sans-serif';
  font-size: 11px;
  font-weight: 400;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  color: #000;
  white-space: nowrap;
}

.c4 {
  margin-top: 16px;
}

.c7 {
  position: absolute;
  right: 16px;
  bottom: 16px;
  width: 40px;
  height: 40px;
  padding: 8px;
  border-radius: 50%;
  background: #fff;
  display: none;
}

.c0:hover .c6 {
  display: block;
}

.c9 {
  position: absolute;
  right: 16px;
  top: 16px;
}

.c0:hover .c8 {
  right: 15px;
  top: 15px;
}

<div
  className="c0 c1"
  data-testid="card-area"
  label="Label"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  shape="square"
  type="detail"
  width="100%"
>
  <div
    className="c2"
    data-testid="card-content"
    shape="square"
  >
    <div
      className="c3"
      data-testid="card-description"
      shape="square"
      type="detail"
    >
      Description
    </div>
    <div
      className="c4"
      data-testid="card-label-container"
    >
      <label
        className="c5"
        data-testid="card-label"
      >
        Label
      </label>
    </div>
  </div>
  <div
    className="c6 c7"
    data-testid="card-detail-hover-arrow"
  >
    <div
      className=""
    >
      <i
        aria-hidden="true"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" width=\\"24px\\" height=\\"24px\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M23.5 12a.75.75 0 01-.75.75h-22a.75.75 0 010-1.5h22a.75.75 0 01.75.75z\\" fill=\\"#000\\"/><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M23.776 11.482c.299.286.299.75 0 1.037L13.052 22.785a.79.79 0 01-1.083 0 .712.712 0 010-1.037L22.151 12 11.969 2.252a.711.711 0 010-1.037.79.79 0 011.083 0l10.724 10.267z\\" fill=\\"#000\\"/></svg>",
          }
        }
        style={
          Object {
            "display": "flex",
          }
        }
      />
    </div>
  </div>
  <div
    className="c8 c9"
    data-testid="card-corner-icon"
  >
    CornerIcon
  </div>
</div>
`;

exports[`Elvis Card Snapshots Type = simple should render correctly circle 1`] = `
.c1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
  overflow: hidden;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 112px;
  max-width: 400px;
  padding: 1px;
  width: 100%;
  height: 100%;
  aspect-ratio: 1/1;
  border-radius: 50%;
  border: 1px solid transparent;
}

.c1:hover {
  border: 2px solid #29d305;
  padding: 0;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 24px;
  padding-right: 24px;
}

.c5 {
  font-family: "Red Hat Text",Verdana,sans-serif;
  font-size: 16px;
  font-weight: 500;
  line-height: 28px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c6 {
  font-family: "Red Hat Text",Verdana,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c4 {
  font-family: "Red Hat Display",Verdana,sans-serif;
  font-size: 44px;
  font-size-mobile: 32px;
  font-weight: 900;
  line-height: 53px;
  line-height-mobile: 34px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  white-space: nowrap;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
}

.c0:hover .c3 {
  -webkit-transform: scale(1.1);
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div
  className="c0 c1"
  data-testid="card-area"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  shape="circle"
  type="simple"
  width="100%"
>
  <div
    className="c2"
    data-testid="card-content"
    shape="circle"
  >
    <div
      className="c3 c4"
      data-testid="card-icon"
    >
      Icon
    </div>
    <div
      className="c5"
      data-testid="card-header"
      shape="circle"
      type="simple"
    >
      Header
    </div>
    <div
      className="c6"
      data-testid="card-description"
      shape="circle"
      type="simple"
    >
      Description
    </div>
  </div>
</div>
`;

exports[`Elvis Card Snapshots Type = simple should render correctly square 1`] = `
.c1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #fff;
  overflow: hidden;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 112px;
  max-width: 400px;
  padding: 24px;
  width: 100%;
  height: 100%;
  border-radius: 8px;
  border: 1px solid #e9e9e9;
}

.c1:hover {
  border: 2px solid #29d305;
  padding: 23px;
  cursor: pointer;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  padding-left: 0;
  padding-right: 0;
}

.c7 {
  font-family: "Red Hat Text",Verdana,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c8 {
  font-family: "Red Hat Text",Verdana,sans-serif;
  font-size: 11px;
  font-weight: 400;
  line-height: 14px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c6 {
  font-family: "Red Hat Display",Verdana,sans-serif;
  font-size: 44px;
  font-size-mobile: 32px;
  font-weight: 900;
  line-height: 53px;
  line-height-mobile: 34px;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  font-style: unset;
  text-transform: unset;
  color: inherit;
  text-align: center;
  color: #000;
  white-space: nowrap;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
}

.c0:hover .c5 {
  -webkit-transform: scale(1.1);
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c3 {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  border-top: 4px solid #ffa000;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
}

.c0:hover .c2 {
  display: none;
}

<div
  className="c0 c1"
  data-testid="card-area"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  shape="square"
  type="simple"
  width="100%"
>
  <div
    className="c2 c3"
    data-testid="card-colored-line"
  />
  <div
    className="c4"
    data-testid="card-content"
    shape="square"
  >
    <div
      className="c5 c6"
      data-testid="card-icon"
    >
      Icon
    </div>
    <div
      className="c7"
      data-testid="card-header"
      shape="square"
      type="simple"
    >
      Header
    </div>
    <div
      className="c8"
      data-testid="card-description"
      shape="square"
      type="simple"
    >
      Description
    </div>
  </div>
</div>
`;
